tools/docker-network/tests/dockerframework.go

```diff
@@ -0,0 +1,339 @@
+}
+
+func NewDockerTestFramework(t *testing.T, validatorNames map[string]string, clientURLs map[string]string, opts ...options.Option[DockerTestFramework]) *DockerTestFramework {
+	return options.Apply(&DockerTestFramework{
+		Testing:                 t,
+		validatorContainerNames: validatorNames,
+		clientURLs:              clientURLs,
+		nodeClients:             make(map[string]*nodeclient.Client),
+		optsWaitForSync:         5 * time.Minute,
+		optsWaitFor:             2 * time.Minute,
+		optsTick:                5 * time.Second,
+	}, opts, func(d *DockerTestFramework) {
+		d.optsProtocolParameterOptions = append(DefaultProtocolParameterOptions, d.optsProtocolParameterOptions...)
+
+		protocolParams := iotago.NewV3ProtocolParameters(d.optsProtocolParameterOptions...)
+		api := iotago.V3API(protocolParams)
+
+		d.logDirectoryPath = createLogDirectory(t.Name())
+		d.snapshotPath = snapshotFilePath
+		d.optsSnapshotOptions = append(DefaultAccountOptions(protocolParams),
+			[]options.Option[snapshotcreator.Options]{
+				snapshotcreator.WithDatabaseVersion(protocol.DatabaseVersion),
+				snapshotcreator.WithFilePath(d.snapshotPath),
+				snapshotcreator.WithProtocolParameters(api.ProtocolParameters()),
+				snapshotcreator.WithRootBlocks(map[iotago.BlockID]iotago.CommitmentID{
+					api.ProtocolParameters().GenesisBlockID(): iotago.NewEmptyCommitment(api).MustID(),
+				}),
+				snapshotcreator.WithGenesisKeyManager(keyManager()),
+			}...)
+
+		err := snapshotcreator.CreateSnapshot(d.optsSnapshotOptions...)
+		if err != nil {
+			panic(fmt.Sprintf("failed to create snapshot: %s", err))
+		}
+	})
+}
+
+func (d *DockerTestFramework) Run() error {
+	go func() {
+		exec.Command("docker", "compose", "up").Run()
+	}()
+
+	// first wait until the nodes are available
+	d.nodeClients = make(map[string]*nodeclient.Client)
+	for name, url := range d.clientURLs {
+		for {
+			client, err := nodeclient.New(url)
+			if err == nil {
+				d.nodeClients[name] = client
+				break
+			}
+
+			time.Sleep(d.optsTick)
+			fmt.Printf("Waiting for node %s to be available...\n", name)
+		}
+	}
+
+	// make sure all nodes are up then we can start dumping logs
+	d.DumpContainerLogsToFiles()
+
+	return nil
+}
+
+func (d *DockerTestFramework) WaitUntilSync() error {
+	fmt.Println("Wait until the nodes are synced...")
+	defer fmt.Println("Wait until the nodes are synced......done")
+
+	d.Eventually(func() error {
+		for name, client := range d.nodeClients {
+			for {
+				synced, err := client.Health(context.TODO())
+				if err != nil {
+					return err
+				}
+
+				if synced {
+					fmt.Println("Node", name, "is synced")
+					break
+				}
+			}
+		}
+
+		return nil
+	}, true)
+
+	return nil
+}
+
+func (d *DockerTestFramework) GetRandomNode() *nodeclient.Client {
+	for _, client := range d.nodeClients {
+		return client
+	}
+
+	return nil
+}
+
+func (d *DockerTestFramework) AssertCommitteeSelection(expectedEpoch iotago.EpochIndex, expectedCommitteeSize int) {
```

Comment:
Maybe rename to `AssertCommitteeSize`? And if we have the validators' accounts, we should probably better check for the exact committee to make sure the expected candidates have been selected